src/windows-emulator/module/module_manager.hpp

```diff
@@ -83,9 +83,15 @@ class module_manager
     callbacks* callbacks_{};
 
     module_map modules_{};
+    mutable module_map::iterator last_module_cache_{modules_.end()};
 
     module_map::iterator get_module(const uint64_t address)
     {
+        if (last_module_cache_ != this->modules_.end() && last_module_cache_->second.contains(address))
```

Comment:
this can break when the last module is unmapped. this leaves a dangling iterator.
why not just fetch the last_module iterator here on the fly?

Reply:
i see. aight, let's keep it this way then. sanitizer hopefully reveals any invalid iterators, in case there ever happen to be any :P